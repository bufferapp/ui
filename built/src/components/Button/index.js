export { default } from './Button';
//# sourceMappingURL=index.js.map