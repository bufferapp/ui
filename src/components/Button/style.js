/* eslint-disable no-confusing-arrow */
import style, { css } from 'styled-components';
import {
  grayDarker,
  blue,
  blueDark,
  blueLighter,
  white,
  grayDark,
  gray,
  grayDefault,
  boxShadow,
  grayLight,
  orangeDarker,
  redDark,
  redDarker,
  redLighter,
  orange as orangeColor,
} from '../style/colors';
import {
  fontFamily,
  fontSize,
  lineHeight,
  fontWeightMedium,
} from '../style/fonts';
import { borderRadius } from '../style/borders';
import { ButtonStyled } from './Button';

export const ButtonWrapperBase = css`
  -webkit-appearance: none;
  align-items: center;
  background-color: ${white};
  border: 0;
  border-radius: ${borderRadius};
  box-sizing: border-box;
  color: ${grayDefault};
  cursor: pointer;
  display: inline-block;
  justify-content: flex-start;
  flex: 0 0 auto;
  font-family: ${fontFamily};
  font-size: ${fontSize};
  font-weight: ${fontWeightMedium};
  line-height: ${lineHeight};
  min-width: 0;
  padding-top: 0;
  padding-bottom: 0;
  position: relative;
  transition-property: background-color, border-color, color;
  transition-duration: 0.1s;
  transition-timing-function: ease-in;
  user-select: none;

  :active {
    top: 1px;
  }
  :focus {
    box-shadow: 0 0 0 3px ${boxShadow};
    outline: 0;
  }
  :hover {
    transition-property: background-color, border-color, color;
    transition-duration: 0.1s;
    transition-timing-function: ease-in;
  }
`;

export const ButtonContainerBase = css`
  -webkit-appearance: none;
  align-items: center;
  cursor: pointer;
  display: flex;
  justify-content: flex-start;
  flex: 0 0 auto;
  user-select: none;
  width: 100%;
  height: 100%;
`;

export const ButtonNestedBase = css`
  align-items: center;
  background: none transparent;
  border: 0;
  border-radius: ${borderRadius};
  color: inherit;
  cursor: pointer;
  display: flex;
  font-family: inherit;
  font-size: inherit;
  font-weight: inherit;
  justify-content: flex-start;
  margin: -1px;
  padding: 0;
  text-overflow: ellipsis;
  white-space: nowrap;
  width: 100%;
`;

/* styles common to all buttons */
export const ButtonBase = css`
  -webkit-appearance: none;
  align-items: center;
  background-color: ${white};
  border: 0;
  border-radius: ${borderRadius};
  color: ${grayDefault};
  cursor: pointer;
  display: flex;
  justify-content: flex-start;
  flex: 0 0 auto;
  font-family: ${fontFamily};
  font-size: ${fontSize};
  font-weight: ${fontWeightMedium};
  line-height: ${lineHeight};
  min-width: 0;
  padding-top: 0;
  padding-bottom: 0;
  position: relative;
  transition-property: background-color, border-color, color;
  transition-duration: 0.1s;
  transition-timing-function: ease-in;
  user-select: none;
  :active {
    top: 1px;
  }
  :focus {
    box-shadow: 0 0 0 3px ${boxShadow};
  }
  :hover {
    transition-property: background-color, border-color, color;
    transition-duration: 0.1s;
    transition-timing-function: ease-in;
  }
`;

/* theme variants */
export const primary = css`
  background-color: ${blue};
  color: ${white};
  border: 1px solid ${blue};
  :hover {
    background-color: ${blueDark};
    border-color: ${blueDark};
    color: ${white};
  }
`;

export const secondary = css`
  background-color: ${white};
  color: ${grayDark};
  border: 1px solid ${gray};

  :hover {
    color: ${grayDarker};
    border-color: ${grayDark};
  }

  :focus {
    border-color: ${blue};
    color: ${grayDarker};
  }
`;

export const danger = css`
  background-color: ${props => (props.loading ? redDarker : redDark)};
  color: ${white};
  border-color: ${redDarker};

  :hover {
    background-color: ${redDarker};
  }

  :focus {
    box-shadow: 0 0 0 3px ${redLighter};
    background-color: ${redDark};
  }
`;

export const link = css`
  background-color: transparent;
  color: ${blue};
  :hover {
    color: ${blueDark};
  }
`;
export const text = css`
  background-color: transparent;
  :hover {
    color: ${grayDarker};
  }
`;

export const orange = css`
  background-color: ${orangeDarker};
  color: ${white};
  border: 1px solid ${orangeDarker};
  :hover {
    background-color: ${orangeColor};
    border-color: ${orangeColor};
    color: ${white};
  }
`;

/* size variants */
export const small = css`
  padding-left: 8px;
  padding-right: 8px;
  height: 32px;
`;
export const medium = css`
  padding-right: 16px;
  padding-left: 16px;
  height: 40px;
`;
export const large = css`
  padding-right: 16px;
  padding-left: 16px;
  height: 48px;
`;

const Sizes = {
  small, medium, large
}

/* state variants */
export const disabled = css`
  background-color: ${grayLight};
  cursor: not-allowed;
  border: 1px solid ${grayLight};
`;

export const fullWidth = css`
  width: 100%;
  justify-content: center;
`;

export const ButtonSelect = style.div`
  :before {
    background-color: ${props => (props.disabled ? gray : blueLighter)};
    color: ${props => (props.disabled ? gray : blueLighter)};
    content: "";
    height: 24px;
    left: 0;
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
    width: 1px;
  }

  ${props => Sizes[props.size] || Sizes.medium};
  align-items: center;
  display: flex;
  justify-content: center;
  box-sizing: border-box;
  ${props => props.size === 'small' ? 'padding: 0 6px'  : 'padding 0 10px'};
  position: relative;
  width: 100%;

  ${ButtonStyled}:hover & {
    background-color: ${blueDark};
    color: ${white};
  }
  ${ButtonStyled}:focus & {
    border-color: ${blue};
      color: ${grayDarker};
  }
`;

export const ButtonArrow = style.div`
  padding-left: 5px;
  display: flex;
  margin-right: -4px;
`;

export const ButtonLabel = style.div`
  margin-left: ${props => (props.hasIcon && !props.iconEnd ? '5px' : '0px')};
  margin-right: ${props => (props.hasIcon && props.iconEnd ? '5px' : '0px')};
`;
