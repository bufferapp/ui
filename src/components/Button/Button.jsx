import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import * as Styles from './style';
import ChevronDown from '../Icon/Icons/ChevronDown';
import Select from '../Select/Select';

/*
Since buttons keep their own inline-block display type, we can only imitate this by using a wrapper (with
 `display: inline-block`) an internal container (with `display: flex`), and then the actual contents (including real
  buttons
 */
export const ButtonWrapperStyled = styled.div`
  ${Styles.ButtonWrapperBase};
  ${props => Styles[props.disabled ? 'disabled' : props.type]};
  ${props => (props.fullWidth ? Styles.fullWidth : '')};
`;

export const ButtonContainerStyled = styled.div`
  ${Styles.ButtonContainerBase};
`;

export const ButtonStyled = styled.button`
  ${Styles.ButtonNestedBase};
  ${props => Styles[props.size]};
  ${props => (props.disabled ? Styles.disabled : '')};
  ${props => (props.fullWidth ? Styles.fullWidth : '')};
  ${props => (props.loading && props.type === 'primary' ? Styles.primaryLoading : '')};
`;

const Loading = styled.img`
  width: 24px;
  margin-left: 10px;
`;

const VisuallyHiddenLabel = styled.span`
  position: absolute;
  overflow: hidden;
  clip: rect(0 0 0 0);
  height: 1px;
  width: 1px;
  margin: -1px;
  padding: 0;
  border: 0;
`;

/** All buttons, including text and split-buttons, follow the same core principles in dimensions, padding, and font sizes.
 * Combined with simple modifiers, they can be changed in size and appearance.  */
const Button = ({
  disabled,
  onClick,
  type,
  size,
  label,
  isSplit,
  loading,
  icon,
  iconEnd,
  hasIconOnly,
  isSelect,
  items,
  selectPosition,
  onSelectClick,
  fullWidth,
  tooltip,
  ref,
  hideSearch,
  className,
  children,
  ...props
}) =>  (
  <ButtonWrapperStyled
    className={className}
    disabled={disabled}
    type={type}
    fullWidth={fullWidth}
    $loading={loading}
  >
    <ButtonContainerStyled>
      <ButtonStyled
        onClick={!disabled ? onClick : undefined}
        disabled={disabled}
        isSplit={isSplit}
        icon={icon}
        hasIconOnly={hasIconOnly}
        data-tip={tooltip}
        ref={ref}
        aria-haspopup="false"
        size={size}
        fullWidth={fullWidth}
        loading={loading}
        type={type}
        {...props}
      >
        {!iconEnd && icon}
        {hasIconOnly && <VisuallyHiddenLabel>{label}</VisuallyHiddenLabel>}
        {!hasIconOnly && (
          <Styles.ButtonLabel hasIcon={!!icon} iconEnd={!!iconEnd}>
            {label}
          </Styles.ButtonLabel>
        )}
        {iconEnd && icon}

        {isSelect && (type === 'primary' || type === 'secondary') && (
          <Styles.ButtonArrow>
            <ChevronDown
              color={type === 'primary' ? 'white' : 'grayDark'}
              size={size}
              isChevron
            />
          </Styles.ButtonArrow>
        )}
        {loading && <Loading src="./images/loading-gray.svg" alt="loading" />}
      </ButtonStyled>
      {isSplit &&
        (type === 'primary' || type === 'secondary') &&
        (children ? (
          React.Children.map(children, child => React.cloneElement(child, {}))
        ) : (
          <Select
            onSelectClick={onSelectClick}
            items={items}
            type={type}
            isSplit
            yPosition={selectPosition}
            xPosition="right"
            disabled={disabled}
            hideSearch={hideSearch}
          />
        ))}
    </ButtonContainerStyled>
  </ButtonWrapperStyled>
  );

Button.propTypes = {
  /** Is the button disabled */
  disabled: PropTypes.bool,

  /** Size of the Button */
  size: PropTypes.oneOf(['small', 'large', 'medium']),

  /** OnClick handler */
  onClick: PropTypes.func.isRequired,

  /** Button label */
  label: PropTypes.string,

  /** Type of button */
  type: PropTypes.oneOf([
    'primary',
    'secondary',
    'text',
    'error',
    'danger',
  ]),

  /** Is the Button Split  */
  isSplit: PropTypes.bool,

  /** Is this the Select button with chevron */
  isSelect: PropTypes.bool,

  /** Is the Button Loading  */
  loading: PropTypes.bool,

  /** Does the button have only an icon and no label */
  hasIconOnly: PropTypes.bool,

  /** Icon to show with the label */
  icon: PropTypes.node,

  /** Icon to show with the label */
  iconEnd: PropTypes.bool,

  /** Items to display in the Split Button popup */
  items: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      title: PropTypes.string,
    })
  ),

  /** Child element(s) to use for custom Split Button child content */
  children: PropTypes.element,

  /** Position of the Select popup */
  selectPosition: PropTypes.oneOf(['top', 'bottom']),

  /** Function to call on Split Button selected item click */
  onSelectClick: PropTypes.func,

  /** Is the button the full width of the parent container */
  fullWidth: PropTypes.bool,

  /** Tooltip to show on the component */
  tooltip: PropTypes.string,

  /** The prop to get the DOM element of the Button */
  ref: PropTypes.node,

  /** Is search hidden */
  hideSearch: PropTypes.bool,

  /** class passed by the dom element */
  className: PropTypes.string,
};

Button.defaultProps = {
  disabled: false,
  isSplit: undefined,
  loading: false,
  size: 'medium',
  type: 'secondary',
  label: undefined,
  hasIconOnly: false,
  icon: undefined,
  iconEnd: false,
  isSelect: undefined,
  items: undefined,
  selectPosition: 'bottom',
  onSelectClick: undefined,
  fullWidth: false,
  tooltip: undefined,
  ref: undefined,
  hideSearch: true,
  className: undefined,
  children: undefined,
};

export default Button;
