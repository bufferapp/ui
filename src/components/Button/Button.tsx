import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import * as Styles from './style'
import ChevronDown from '../Icon/Icons/ChevronDown'
import Select from '../Select/Select'

/*
Since buttons keep their own inline-block display type, we can only imitate this by using a wrapper (with
 `display: inline-block`) an internal container (with `display: flex`), and then the actual contents (including real
  buttons
 */

export const ButtonWrapperStyled = styled.div`
  ${Styles.ButtonWrapperBase};
  ${(props) =>
    // @ts-expect-error TS(7053) FIXME: Element implicitly has an 'any' type because expre... Remove this comment to see the full error message
    Styles[[props.type, props.disabled ? 'Disabled' : ''].join('')]};
  ${(props): string =>  
// @ts-expect-error TS(2322) FIXME: Type 'FlattenSimpleInterpolation | ""' is not assi... Remove this comment to see the full error message
 (props.fullWidth ? Styles.fullWidth : '')};
`

export const ButtonContainerStyled = styled.div`
  ${Styles.ButtonContainerBase};
`

export const ButtonStyled = styled.button`
  ${Styles.ButtonNestedBase};
  ${(props) =>
    // @ts-expect-error TS(7053) FIXME: Element implicitly has an 'any' type because expre... Remove this comment to see the full error message
    Styles[props.size]};
  ${(props) =>
    // @ts-expect-error TS(7053) FIXME: Element implicitly has an 'any' type because expre... Remove this comment to see the full error message
    Styles[props.disabled ? 'disabled' : '']};
  ${(props): string =>  
// @ts-expect-error TS(2322) FIXME: Type 'FlattenSimpleInterpolation | ""' is not assi... Remove this comment to see the full error message
 (props.fullWidth ? Styles.fullWidth : '')};
`

const Loading = styled.img`
  width: 24px;
  margin-left: 10px;
`

const VisuallyHiddenLabel = styled.span`
  position: absolute;
  overflow: hidden;
  clip: rect(0 0 0 0);
  height: 1px;
  width: 1px;
  margin: -1px;
  padding: 0;
  border: 0;
`

/** All buttons, including text and split-buttons, follow the same core principles in dimensions, padding, and font sizes.
 * Combined with simple modifiers, they can be changed in size and appearance.  */
const Button = ({
  // @ts-expect-error TS(7031) FIXME: Binding element 'disabled' implicitly has an 'any'... Remove this comment to see the full error message
  disabled,
  // @ts-expect-error TS(7031) FIXME: Binding element 'onClick' implicitly has an 'any' ... Remove this comment to see the full error message
  onClick,
  // @ts-expect-error TS(7031) FIXME: Binding element 'type' implicitly has an 'any' typ... Remove this comment to see the full error message
  type,
  // @ts-expect-error TS(7031) FIXME: Binding element 'size' implicitly has an 'any' typ... Remove this comment to see the full error message
  size,
  // @ts-expect-error TS(7031) FIXME: Binding element 'label' implicitly has an 'any' ty... Remove this comment to see the full error message
  label,
  // @ts-expect-error TS(7031) FIXME: Binding element 'isSplit' implicitly has an 'any' ... Remove this comment to see the full error message
  isSplit,
  // @ts-expect-error TS(7031) FIXME: Binding element 'loading' implicitly has an 'any' ... Remove this comment to see the full error message
  loading,
  // @ts-expect-error TS(7031) FIXME: Binding element 'icon' implicitly has an 'any' typ... Remove this comment to see the full error message
  icon,
  // @ts-expect-error TS(7031) FIXME: Binding element 'iconEnd' implicitly has an 'any' ... Remove this comment to see the full error message
  iconEnd,
  // @ts-expect-error TS(7031) FIXME: Binding element 'hasIconOnly' implicitly has an 'a... Remove this comment to see the full error message
  hasIconOnly,
  // @ts-expect-error TS(7031) FIXME: Binding element 'isSelect' implicitly has an 'any'... Remove this comment to see the full error message
  isSelect,
  // @ts-expect-error TS(7031) FIXME: Binding element 'items' implicitly has an 'any' ty... Remove this comment to see the full error message
  items,
  // @ts-expect-error TS(7031) FIXME: Binding element 'selectPosition' implicitly has an... Remove this comment to see the full error message
  selectPosition,
  // @ts-expect-error TS(7031) FIXME: Binding element 'onSelectClick' implicitly has an ... Remove this comment to see the full error message
  onSelectClick,
  // @ts-expect-error TS(7031) FIXME: Binding element 'fullWidth' implicitly has an 'any... Remove this comment to see the full error message
  fullWidth,
  // @ts-expect-error TS(7031) FIXME: Binding element 'tooltip' implicitly has an 'any' ... Remove this comment to see the full error message
  tooltip,
  // @ts-expect-error TS(7031) FIXME: Binding element 'ref' implicitly has an 'any' type... Remove this comment to see the full error message
  ref,
  // @ts-expect-error TS(7031) FIXME: Binding element 'hideSearch' implicitly has an 'an... Remove this comment to see the full error message
  hideSearch,
  // @ts-expect-error TS(7031) FIXME: Binding element 'textToLeft' implicitly has an 'an... Remove this comment to see the full error message
  textToLeft,
  // @ts-expect-error TS(7031) FIXME: Binding element 'className' implicitly has an 'any... Remove this comment to see the full error message
  className,
  // @ts-expect-error TS(7031) FIXME: Binding element 'children' implicitly has an 'any'... Remove this comment to see the full error message
  children,
  // @ts-expect-error TS(7031) FIXME: Binding element 'onOpen' implicitly has an 'any' t... Remove this comment to see the full error message
  onOpen,
  ...props
}) => (
  <ButtonWrapperStyled
    className={className}
    // @ts-expect-error TS(2769) FIXME: No overload matches this call.
    disabled={disabled}
    type={type}
    fullWidth={fullWidth}
    $loading={loading}
  >
    <ButtonContainerStyled>
      <ButtonStyled
        onClick={!disabled ? onClick : undefined}
        disabled={disabled}
        // @ts-expect-error TS(2769) FIXME: No overload matches this call.
        isSplit={isSplit}
        icon={icon}
        hasIconOnly={hasIconOnly}
        data-tip={tooltip}
        ref={ref}
        aria-haspopup="false"
        size={size}
        fullWidth={fullWidth}
        type={type}
        {...props}
      >
        {!iconEnd && icon}
        {hasIconOnly && <VisuallyHiddenLabel>{label}</VisuallyHiddenLabel>}
        {!hasIconOnly && (
          // @ts-expect-error TS(2769) FIXME: No overload matches this call.
          <Styles.ButtonLabel hasIcon={!!icon} iconEnd={!!iconEnd}>
            {label}
          </Styles.ButtonLabel>
        )}
        {iconEnd && icon}

        {isSelect && (type === 'primary' || type === 'secondary') && (
          // @ts-expect-error TS(2769) FIXME: No overload matches this call.
          <Styles.ButtonArrow textToLeft={textToLeft}>
            <ChevronDown
              color={type === 'primary' ? 'white' : 'grayDark'}
              size={size}
              isChevron
            />
          </Styles.ButtonArrow>
        )}

        {loading && <Loading src="./images/loading-gray.svg" alt="loading" />}
      </ButtonStyled>
      {isSplit &&
        (type === 'primary' || type === 'secondary') &&
        (children ? (
          React.Children.map(children, (child) => React.cloneElement(child, {}))
        ) : (
          <Select
            // @ts-expect-error TS(2769) FIXME: No overload matches this call.
            onSelectClick={onSelectClick}
            items={items}
            type={type}
            size={size}
            isSplit
            yPosition={selectPosition}
            xPosition="right"
            disabled={disabled}
            hideSearch={hideSearch}
            onOpen={onOpen}
          />
        ))}
    </ButtonContainerStyled>
  </ButtonWrapperStyled>
)

Button.propTypes = {
  /** Is the button disabled */
  disabled: PropTypes.bool,

  /** Size of the Button */
  size: PropTypes.oneOf(['small', 'large', 'medium']),

  /** OnClick handler */
  onClick: PropTypes.func.isRequired,

  /** Button label */
  label: PropTypes.string,

  /** Type of button */
  type: PropTypes.oneOf([
    'primary',
    'secondary',
    'text',
    'error',
    'danger',
    'orange',
  ]),

  /** Is the Button Split  */
  isSplit: PropTypes.bool,

  /** Is this the Select button with chevron */
  isSelect: PropTypes.bool,

  /** Is the Button Loading  */
  loading: PropTypes.bool,

  /** Does the button have only an icon and no label */
  hasIconOnly: PropTypes.bool,

  /** Icon to show with the label */
  icon: PropTypes.node,

  /** Icon to show with the label */
  iconEnd: PropTypes.bool,

  /** Items to display in the Split Button popup */
  items: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      title: PropTypes.string,
    }),
  ),

  /** Child element(s) to use for custom Split Button child content */
  children: PropTypes.element,

  /** Position of the Select popup */
  selectPosition: PropTypes.oneOf(['top', 'bottom']),

  /** Function to call on Split Button selected item click */
  onSelectClick: PropTypes.func,

  /** Is the button the full width of the parent container */
  fullWidth: PropTypes.bool,

  /** Tooltip to show on the component */
  tooltip: PropTypes.string,

  /** The prop to get the DOM element of the Button */
  ref: PropTypes.node,

  /** Is search hidden */
  hideSearch: PropTypes.bool,

  /** class passed by the dom element */
  className: PropTypes.string,

  /** onOpen function to fire when the Dropdown menu is open */
  onOpen: PropTypes.func,
}

Button.defaultProps = {
  disabled: false,
  isSplit: undefined,
  loading: false,
  size: 'medium',
  type: 'secondary',
  label: undefined,
  hasIconOnly: false,
  icon: undefined,
  iconEnd: false,
  isSelect: undefined,
  items: undefined,
  selectPosition: 'bottom',
  onSelectClick: undefined,
  fullWidth: false,
  tooltip: undefined,
  ref: undefined,
  hideSearch: true,
  className: undefined,
  children: undefined,
  onOpen: null,
}

export default Button
