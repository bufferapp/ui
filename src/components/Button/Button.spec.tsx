/* eslint-disable react/jsx-filename-extension */
import React from 'react'
import snap from 'jest-auto-snapshots'
import renderer from 'react-test-renderer'
import Button from './Button'
import 'jest-styled-components'

snap(Button, './Button.jsx')

test('Button renders styles properly', () => {
  const tree = renderer
    .create(<Button type="primary" onClick={() => {}} label="Primary" />)
    .toJSON()
  expect(tree).toHaveStyleRule('color', '#FFFFFF')
  expect(tree).toHaveStyleRule('cursor', 'pointer')
})

test('Disabled button is disabled', () => {
  const tree = renderer
    .create(
      <Button type="primary" onClick={() => {}} label="Primary" disabled />,
    )
    .toJSON()
  expect(tree).toHaveStyleRule('background-color', '#c6cbe4')
  expect(tree).toHaveStyleRule('cursor', 'not-allowed')
})

test('Primary default loading button has a darker blue background', () => {
  const tree = renderer
    .create(
      <Button type="primary" onClick={() => {}} label="Primary" loading />,
    )
    .toJSON()
  expect(tree).toHaveStyleRule('background-color', '#121E66')
})
