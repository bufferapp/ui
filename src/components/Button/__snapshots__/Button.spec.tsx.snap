// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > Button Matches snapshot when array prop "items" has one item: "[object Object]" 1`] = `
<styled.div
  $loading={true}
  className="jest-auto-snapshots String Fixture"
  disabled={true}
  fullWidth={true}
  type="primary"
>
  <styled.div>
    <styled.button
      aria-haspopup="false"
      data-tip="jest-auto-snapshots String Fixture"
      disabled={true}
      fullWidth={true}
      hasIconOnly={true}
      icon={<NodeFixture />}
      isSplit={true}
      size="small"
      type="primary"
    >
      <styled.span>
        jest-auto-snapshots String Fixture
      </styled.span>
      <NodeFixture />
      <styled.div>
        <ChevronDownIcon
          color="white"
          isChevron={true}
          size="small"
        />
      </styled.div>
      <styled.img
        alt="loading"
        src="./images/loading-gray.svg"
      />
    </styled.button>
    <NodeFixture />
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > Button Matches snapshot when array prop "items" is an empty array: "[]" 1`] = `
<styled.div
  $loading={true}
  className="jest-auto-snapshots String Fixture"
  disabled={true}
  fullWidth={true}
  type="primary"
>
  <styled.div>
    <styled.button
      aria-haspopup="false"
      data-tip="jest-auto-snapshots String Fixture"
      disabled={true}
      fullWidth={true}
      hasIconOnly={true}
      icon={<NodeFixture />}
      isSplit={true}
      size="small"
      type="primary"
    >
      <styled.span>
        jest-auto-snapshots String Fixture
      </styled.span>
      <NodeFixture />
      <styled.div>
        <ChevronDownIcon
          color="white"
          isChevron={true}
          size="small"
        />
      </styled.div>
      <styled.img
        alt="loading"
        src="./images/loading-gray.svg"
      />
    </styled.button>
    <NodeFixture />
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > Button Matches snapshot when boolean prop "disabled" is set to: "false" 1`] = `
<styled.div
  $loading={true}
  className="jest-auto-snapshots String Fixture"
  disabled={false}
  fullWidth={true}
  type="primary"
>
  <styled.div>
    <styled.button
      aria-haspopup="false"
      data-tip="jest-auto-snapshots String Fixture"
      disabled={false}
      fullWidth={true}
      hasIconOnly={true}
      icon={<NodeFixture />}
      isSplit={true}
      onClick={[MockFunction]}
      size="small"
      type="primary"
    >
      <styled.span>
        jest-auto-snapshots String Fixture
      </styled.span>
      <NodeFixture />
      <styled.div>
        <ChevronDownIcon
          color="white"
          isChevron={true}
          size="small"
        />
      </styled.div>
      <styled.img
        alt="loading"
        src="./images/loading-gray.svg"
      />
    </styled.button>
    <NodeFixture />
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > Button Matches snapshot when boolean prop "fullWidth" is set to: "false" 1`] = `
<styled.div
  $loading={true}
  className="jest-auto-snapshots String Fixture"
  disabled={true}
  fullWidth={false}
  type="primary"
>
  <styled.div>
    <styled.button
      aria-haspopup="false"
      data-tip="jest-auto-snapshots String Fixture"
      disabled={true}
      fullWidth={false}
      hasIconOnly={true}
      icon={<NodeFixture />}
      isSplit={true}
      size="small"
      type="primary"
    >
      <styled.span>
        jest-auto-snapshots String Fixture
      </styled.span>
      <NodeFixture />
      <styled.div>
        <ChevronDownIcon
          color="white"
          isChevron={true}
          size="small"
        />
      </styled.div>
      <styled.img
        alt="loading"
        src="./images/loading-gray.svg"
      />
    </styled.button>
    <NodeFixture />
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > Button Matches snapshot when boolean prop "hasIconOnly" is set to: "false" 1`] = `
<styled.div
  $loading={true}
  className="jest-auto-snapshots String Fixture"
  disabled={true}
  fullWidth={true}
  type="primary"
>
  <styled.div>
    <styled.button
      aria-haspopup="false"
      data-tip="jest-auto-snapshots String Fixture"
      disabled={true}
      fullWidth={true}
      hasIconOnly={false}
      icon={<NodeFixture />}
      isSplit={true}
      size="small"
      type="primary"
    >
      <styled.div
        hasIcon={true}
        iconEnd={true}
      >
        jest-auto-snapshots String Fixture
      </styled.div>
      <NodeFixture />
      <styled.div>
        <ChevronDownIcon
          color="white"
          isChevron={true}
          size="small"
        />
      </styled.div>
      <styled.img
        alt="loading"
        src="./images/loading-gray.svg"
      />
    </styled.button>
    <NodeFixture />
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > Button Matches snapshot when boolean prop "hideSearch" is set to: "false" 1`] = `
<styled.div
  $loading={true}
  className="jest-auto-snapshots String Fixture"
  disabled={true}
  fullWidth={true}
  type="primary"
>
  <styled.div>
    <styled.button
      aria-haspopup="false"
      data-tip="jest-auto-snapshots String Fixture"
      disabled={true}
      fullWidth={true}
      hasIconOnly={true}
      icon={<NodeFixture />}
      isSplit={true}
      size="small"
      type="primary"
    >
      <styled.span>
        jest-auto-snapshots String Fixture
      </styled.span>
      <NodeFixture />
      <styled.div>
        <ChevronDownIcon
          color="white"
          isChevron={true}
          size="small"
        />
      </styled.div>
      <styled.img
        alt="loading"
        src="./images/loading-gray.svg"
      />
    </styled.button>
    <NodeFixture />
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > Button Matches snapshot when boolean prop "iconEnd" is set to: "false" 1`] = `
<styled.div
  $loading={true}
  className="jest-auto-snapshots String Fixture"
  disabled={true}
  fullWidth={true}
  type="primary"
>
  <styled.div>
    <styled.button
      aria-haspopup="false"
      data-tip="jest-auto-snapshots String Fixture"
      disabled={true}
      fullWidth={true}
      hasIconOnly={true}
      icon={<NodeFixture />}
      isSplit={true}
      size="small"
      type="primary"
    >
      <NodeFixture />
      <styled.span>
        jest-auto-snapshots String Fixture
      </styled.span>
      <styled.div>
        <ChevronDownIcon
          color="white"
          isChevron={true}
          size="small"
        />
      </styled.div>
      <styled.img
        alt="loading"
        src="./images/loading-gray.svg"
      />
    </styled.button>
    <NodeFixture />
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > Button Matches snapshot when boolean prop "isSelect" is set to: "false" 1`] = `
<styled.div
  $loading={true}
  className="jest-auto-snapshots String Fixture"
  disabled={true}
  fullWidth={true}
  type="primary"
>
  <styled.div>
    <styled.button
      aria-haspopup="false"
      data-tip="jest-auto-snapshots String Fixture"
      disabled={true}
      fullWidth={true}
      hasIconOnly={true}
      icon={<NodeFixture />}
      isSplit={true}
      size="small"
      type="primary"
    >
      <styled.span>
        jest-auto-snapshots String Fixture
      </styled.span>
      <NodeFixture />
      <styled.img
        alt="loading"
        src="./images/loading-gray.svg"
      />
    </styled.button>
    <NodeFixture />
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > Button Matches snapshot when boolean prop "isSplit" is set to: "false" 1`] = `
<styled.div
  $loading={true}
  className="jest-auto-snapshots String Fixture"
  disabled={true}
  fullWidth={true}
  type="primary"
>
  <styled.div>
    <styled.button
      aria-haspopup="false"
      data-tip="jest-auto-snapshots String Fixture"
      disabled={true}
      fullWidth={true}
      hasIconOnly={true}
      icon={<NodeFixture />}
      isSplit={false}
      size="small"
      type="primary"
    >
      <styled.span>
        jest-auto-snapshots String Fixture
      </styled.span>
      <NodeFixture />
      <styled.div>
        <ChevronDownIcon
          color="white"
          isChevron={true}
          size="small"
        />
      </styled.div>
      <styled.img
        alt="loading"
        src="./images/loading-gray.svg"
      />
    </styled.button>
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > Button Matches snapshot when boolean prop "loading" is set to: "false" 1`] = `
<styled.div
  $loading={false}
  className="jest-auto-snapshots String Fixture"
  disabled={true}
  fullWidth={true}
  type="primary"
>
  <styled.div>
    <styled.button
      aria-haspopup="false"
      data-tip="jest-auto-snapshots String Fixture"
      disabled={true}
      fullWidth={true}
      hasIconOnly={true}
      icon={<NodeFixture />}
      isSplit={true}
      size="small"
      type="primary"
    >
      <styled.span>
        jest-auto-snapshots String Fixture
      </styled.span>
      <NodeFixture />
      <styled.div>
        <ChevronDownIcon
          color="white"
          isChevron={true}
          size="small"
        />
      </styled.div>
    </styled.button>
    <NodeFixture />
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > Button Matches snapshot when passed all props 1`] = `
<styled.div
  $loading={true}
  className="jest-auto-snapshots String Fixture"
  disabled={true}
  fullWidth={true}
  type="primary"
>
  <styled.div>
    <styled.button
      aria-haspopup="false"
      data-tip="jest-auto-snapshots String Fixture"
      disabled={true}
      fullWidth={true}
      hasIconOnly={true}
      icon={<NodeFixture />}
      isSplit={true}
      size="small"
      type="primary"
    >
      <styled.span>
        jest-auto-snapshots String Fixture
      </styled.span>
      <NodeFixture />
      <styled.div>
        <ChevronDownIcon
          color="white"
          isChevron={true}
          size="small"
        />
      </styled.div>
      <styled.img
        alt="loading"
        src="./images/loading-gray.svg"
      />
    </styled.button>
    <NodeFixture />
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > Button Matches snapshot when passed only required props 1`] = `
<styled.div
  $loading={false}
  disabled={false}
  fullWidth={false}
  type="secondary"
>
  <styled.div>
    <styled.button
      aria-haspopup="false"
      disabled={false}
      fullWidth={false}
      hasIconOnly={false}
      onClick={[MockFunction]}
      size="medium"
      type="secondary"
    >
      <styled.div
        hasIcon={false}
        iconEnd={false}
      />
    </styled.button>
  </styled.div>
</styled.div>
`;
